clang/lib/Driver/Driver.cpp

```diff
@@ -5105,6 +5105,60 @@ class OffloadingActionBuilder final {
       }
     }
 
+    // Return whether to use native bfloat16 library.
+    bool selectBfloatLibs(const ToolChain *TC, bool &useNative) {
+      bool needLibs = false;
+
+      const OptTable &Opts = C.getDriver().getOpts();
+      const char *TargetOpt = nullptr;
+      const char *DeviceOpt = nullptr;
+      for (auto *A : Args) {
+        llvm::Triple *TargetBE = nullptr;
+
+        auto GetTripleIt = [&, this](llvm::StringRef Triple) {
+          llvm::Triple TargetTriple{Triple};
+          auto TripleIt = llvm::find_if(SYCLTripleList, [&](auto &SYCLTriple) {
+            return SYCLTriple == TargetTriple;
+          });
+          return TripleIt != SYCLTripleList.end() ? &*TripleIt : nullptr;
+        };
+
+        if (A->getOption().matches(options::OPT_fsycl_targets_EQ)) {
+          // Passing arg: -fsycl-targets=<targets>.
+          needLibs =
+              TC->getTriple().getSubArch() != llvm::Triple::SPIRSubArch_fpga;
```

Comment:
I'm just curious how this works.  It seems like this will set `needLibs` to `true` for "-fsycl-targets=spirv64".  That's not the case?

I know we have a test that verifies that the bfloat16 libs are not linked in this case, and I assume the test passes.  However, I'm curious how it is that this code works.

Reply:
It was not working as expected. The test had an error. I have tightened up the checking. The implementation also needed a change.
Now "-fsycl-targets=spirv64" is recognized as JIT and no libs are added.